prow/spyglass/lenses/buildlog/buildlog.ts

```diff
@@ -46,16 +46,45 @@ function ansiToHTML(orig: string): string {
   });
 }
 
-async function replaceElementWithContent(element: HTMLDivElement) {
+interface ArtifactRequest {
+  artifact: string;
+  bottom: number;
+  length: number;
+  offset: number;
+  startLine: number;
+  top: number;
+}
+
+async function replaceElementWithContent(element: HTMLDivElement, top: number, bottom: number) {
+  // +length => make it a number (as opposed to a string)
+  // length! => I know this value is non-null
```

Comment:
Style: This isn't safe; should either be a check, or explain that it can't be null + why. (https://google.github.io/styleguide/tsguide.html#type-and-non-nullability-assertions)

Reply:
Done-ish...

I think the idea here (note that this was `+offset!` etc before, is that this data is tightly coupled with how its defined in template.html, like a proto might be. So added a comment and a TODO to make this more robust.